project_internal_access_from_portal/tests/test_portal_internal_access.py

```
@@ -0,0 +1,57 @@
+        self.project_pigs.message_subscribe([self.portal.partner_id.id])
+        _ = self.project_pigs.with_user(self.portal).name
+
+    def test_project_write_unlink_forbidden(self):
+        """Portal user cannot write or unlink at any time"""
+        with self.assertRaises(AccessError):
+            self.project_pigs.with_user(self.portal).write({'name': 'New'})
+        self.project_pigs.message_subscribe([self.portal.partner_id.id])
+        with self.assertRaises(AccessError):
+            self.project_pigs.with_user(self.portal).unlink()
+
+    def test_task_access_control(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_task_access_control`

Add return type annotation: `None`

(ANN201)